src/overlays/actors/ovl_En_Dai/z_en_dai.c

```diff
@@ -22,58 +43,697 @@ const ActorInit En_Dai_InitVars = {
+        this->unk_A6C = 1;
+    }
+}
+
+void func_80B3EF90(EnDai* this, GlobalContext* globalCtx) {
+    if (func_8010BF58(&this->actor, globalCtx, D_80B3FC8C, 0, &this->unk_1D0)) {
+        func_8013AED4(&this->unk_1CE, 3, 7);
+        this->unk_1D0 = 0;
+        this->actionFunc = func_80B3F00C;
+    } else {
+        Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 0x2AA8);
+    }
+}
+
+void func_80B3F00C(EnDai* this, GlobalContext* globalCtx) {
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.world.rot.y, 3, 0x2AA8);
+}
+
+void func_80B3F044(EnDai* this, GlobalContext* globalCtx) {
+    static s32 D_80B3FE38[] = {
+        0x00000000, 0x00000000, 0x00000006, 0x00000007, 0x00000008,
+    };
+
+    s32 sp2C = 0;
+    s32 sp28;
+
+    if (func_800EE29C(globalCtx, 0x1D8)) {
+        sp2C = func_800EE200(globalCtx, 0x1D8);
+        sp28 = globalCtx->csCtx.npcActions[sp2C]->unk0;
+        if (this->unk_1CC != (u8)sp28) {
+            func_80B3E5DC(this, D_80B3FE38[sp28]);
+            switch (sp28) {
+                case 1:
+                    this->unk_1E0 = 0.0f;
+                    break;
+
+                case 2:
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EV_ROLL_AND_FALL);
+                    break;
+
+                case 3:
+                    this->unk_1CE &= ~0x80;
+                    break;
+            }
+        }
+        this->unk_1CC = sp28;
+    }
+
+    switch (this->unk_1CC) {
+        case 1:
+            this->unk_1E0 += 1.0f;
+            if (this->unk_1E0 < 30.0f) {
+                this->unk_1CD = (this->unk_1E0 / 30.0f) * 255.0f;
+            } else {
+                this->unk_1CD = 0xFF;
+            }
+            break;
+
+        case 2:
+            if (globalCtx->csCtx.frames == 0x168) {
+                Audio_PlayActorSound2(&this->actor, 0x30FE);
+            }
+            if (func_801378B8(&this->skelAnime, 43.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_GORON_BOUND_1);
+            }
+            this->unk_1D6 = func_80B3E8BC(this, this->unk_1CC);
+            break;
+
+        case 3:
+            if (func_801378B8(&this->skelAnime, 6.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIGOLON_SLEEP1);
+            }
+            this->unk_1D6 = func_80B3E8BC(this, this->unk_1CC);
+            break;
+
+        case 4:
+            if (func_801378B8(&this->skelAnime, 30.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIGOLON_SLEEP2);
+            }
+            if (func_801378B8(&this->skelAnime, 35.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_GORON_BOUND_0);
+            }
+            this->unk_1D6 = func_80B3E8BC(this, this->unk_1CC);
+            break;
+    }
+
+    func_800EDF24(&this->actor, globalCtx, sp2C);
+}
+
+void EnDai_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDai* this = THIS;
+    u16 temp_t7;
```

Comment:
This temp is unused in this function, so I'd suggest either getting rid of this (or changing it to a pad if needed for stack)